epsilon_greedy.go

```diff
@@ -143,45 +140,55 @@ func (p *epsilonGreedyHostPool) getEpsilonGreedy() string {
 		return p.getRoundRobin()
 	}
 
-	// calculate values for each host in the 0..1 range (but not ormalized)
-	var possibleHosts []*hostEntry
+	// here we are exploiting
+	var (
+		totalScore            float64
+		candidateEpsilonHosts = make([]*hostEntry, 0, 256)
+		candidateScores       = make([]float64, 0, 256)
+	)
+
 	now := time.Now()
-	var sumValues float64
 	for _, h := range p.hostList {
-		if h.canTryHost(now) {
-			v := h.getWeightedAverageResponseTime()
-			if v > 0 {
-				ev := p.CalcValueFromAvgResponseTime(v)
-				h.epsilonValue = ev
-				sumValues += ev
-				possibleHosts = append(possibleHosts, h)
-			}
+		if !h.canTryHost(now) {
+			continue
 		}
-	}
 
-	if len(possibleHosts) != 0 {
-		// now normalize to the 0..1 range to get a percentage
-		for _, h := range possibleHosts {
-			h.epsilonPercentage = h.epsilonValue / sumValues
+		hostResponseTime := h.getWeightedAverageResponseTime()
+		if hostResponseTime <= 0 {
+			continue
 		}
 
-		// do a weighted random choice among hosts
-		ceiling := 0.0
-		pickPercentage := rand.Float64()
-		for _, h := range possibleHosts {
-			ceiling += h.epsilonPercentage
-			if pickPercentage <= ceiling {
-				hostToUse = h
-				break
-			}
+		score := 1.0 / hostResponseTime
```

Comment:
is doing `1.0 / hostResponseTime` here effectively hard-coding what was previously in `CalcValueFromAvgResponseTime`? Do we need to remove the `EpsilonValueCalculator` if we are not using it?

Reply:
It should use `CalcValueFromAvgResponseTime`, i've fixed it 👍